homeassistant/components/ads/__init__.py

```diff
@@ -88,8 +94,17 @@
     }
 )
 
+type PLCDataType = (
+    pyads.PLCTYPE_BOOL
+    | pyads.PLCTYPE_BYTE
+    | pyads.PLCTYPE_DINT
+    | pyads.PLCTYPE_INT
+    | pyads.PLCTYPE_UDINT
+    | pyads.PLCTYPE_UINT
+)
+
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
```

Comment:
When we make this async, does it do any blocking IO? The fact that AdsHub is logging "Could not connect" seems to indicate it can happen.